be/src/vec/functions/function_hex.cpp

```diff
@@ -191,9 +191,85 @@ struct HexHLLImpl {
     }
 };
 
+template <typename Impl>
+class FunctionOctVariadic : public IFunction {
+public:
+    static constexpr auto name = "oct";
+
+    static FunctionPtr create() { return std::make_shared<FunctionOctVariadic>(); }
+
+    String get_name() const override { return name; }
+
+    size_t get_number_of_arguments() const override { return 1; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeString>();
+    }
+
+    DataTypes get_variadic_argument_types_impl() const override {
```

Comment:
Will you support the input of string type in the future? just like `hex`. If not, we have no need to make this function variadic and templative.

Reply:
yes.. will raise new PR to support this.